### PR TITLE
Pfc sniper

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

title

## Why It's Good For The Game

shooty bang bang bye bye mosin
## Changelog
:cl: Kuro, MJP and Tiviplus
add: PFC sniper to replace the mosin
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -815,3 +815,77 @@
...
+	if(racked_bolt)
+		to_chat(user, "<span class='notice'>[src] already has a round chambered!</span>")
+		return
+	if(!(cooldown_time < world.time))
+		return
+	return rack_bolt(user)
+
+/obj/item/weapon/gun/rifle/chambered/proc/rack_bolt(mob/user)
+	to_chat(user, "<span class='notice'>You cycle the bolt of the [src], loading in a new round!</span>")
+	cooldown_time = world.time + rack_delay
```
Comment (psykzz) [inline code/modules/projectiles/guns/rifles.dm:885]: Would you replace these with `COOLDOWN_START` instead

File: code/modules/projectiles/guns/rifles.dm [hunk 2]
```diff
@@ -815,3 +815,77 @@
...
+		to_chat(user, "<span class='notice'>[src] does not have a round chambered!</span>")
+		return FALSE
+	return TRUE
+
+/obj/item/weapon/gun/rifle/chambered/unique_action(mob/user)
+	if(racked_bolt)
+		to_chat(user, "<span class='notice'>[src] already has a round chambered!</span>")
+		return
+	if(!(cooldown_time < world.time))
+		return
```
Comment (psykzz) [inline code/modules/projectiles/guns/rifles.dm:880]: Replace this with `COOLDOWN_CHECK`

`COOLDOWN_CHECK(src, COOLDOWN_GUN_CHAMBER)` would work if you add the define

File: code/modules/projectiles/guns/rifles.dm [hunk 3]
```diff
@@ -815,3 +815,77 @@
...
+	var/rack_sound = "tl-127_bolt.ogg"
+	var/racked_bolt = TRUE
+	var/cooldown_time
+
+/obj/item/weapon/gun/rifle/chambered/able_to_fire(mob/user)
+	. = ..()
+	if(!racked_bolt)
+		to_chat(user, "<span class='notice'>[src] does not have a round chambered!</span>")
+		return FALSE
+	return TRUE
```
Comment (Rohesie) [inline code/modules/projectiles/guns/rifles.dm:873]:
```suggestion

```
Shouldn't ignore the parent return.

File: code/modules/projectiles/guns/rifles.dm [hunk 4]
```diff
@@ -815,3 +815,77 @@
...
+	aim_slowdown = 1
+	wield_delay = 1 SECONDS
+
+	var/rack_delay = 7
+	var/rack_sound = "tl-127_bolt.ogg"
+	var/racked_bolt = TRUE
+	var/cooldown_time
+
+/obj/item/weapon/gun/rifle/chambered/able_to_fire(mob/user)
+	. = ..()
```
Comment (Rohesie) [inline code/modules/projectiles/guns/rifles.dm:869]:
```suggestion
	. = ..()
	if(!.)
		return
```

File: code/modules/projectiles/guns/rifles.dm [hunk 5]
```diff
@@ -815,3 +815,77 @@
...
+
+/obj/item/weapon/gun/rifle/chambered/proc/rack_bolt(mob/user)
+	to_chat(user, "<span class='notice'>You cycle the bolt of the [src], loading in a new round!</span>")
+	cooldown_time = world.time + rack_delay
+	racked_bolt = TRUE
+	playsound(loc, rack_sound, 25, 1, 4)
+
+/obj/item/weapon/gun/rifle/chambered/Fire(atom/target, mob/living/user, params, reflex = 0, dual_wield)
+	. = ..()
+	racked_bolt = FALSE
```
Comment (Rohesie) [inline code/modules/projectiles/guns/rifles.dm:891]: Put this in `reload_into_chamber()`

File: code/modules/projectiles/guns/rifles.dm [hunk 6]
```diff
@@ -815,3 +815,77 @@
...
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/bayonetknife,
+						/obj/item/attachable/compensator)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY|GUN_CAN_POINTBLANK
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO)
+	attachable_offset = list("muzzle_x" = 41, "muzzle_y" = 19,"rail_x" = 8, "rail_y" = 21, "under_x" = 37, "under_y" = 16, "stock_x" = 9, "stock_y" = 12)
+	starting_attachment_types = list(/obj/item/attachable/scope/unremovable/tl127,
+									/obj/item/attachable/stock/tl127stock)
+
```
Comment (Hughgent) [inline code/modules/projectiles/guns/rifles.dm:852]:
```suggestion
burst_amount = 0
```
It's inheriting `burst_amount = 3` from the parent rifle object.

**Comment by psykzz** [comment]
From travis
```
code/modules/projectiles/guns/rifles.dm, line 852, column 17:
error: got '{', expected one of: operator, field access, as, 'in', ';'
- 853:1: check for extra indentation at the start of the next line
- fatal error: the parser cannot continue
- constant evaluation will be skipped
```

File: code/modules/projectiles/guns/rifles.dm [hunk 7]
```diff
@@ -815,3 +815,78 @@
...
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/bayonetknife,
+						/obj/item/attachable/compensator)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY|GUN_CAN_POINTBLANK
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO)
+	attachable_offset = list("muzzle_x" = 41, "muzzle_y" = 19,"rail_x" = 8, "rail_y" = 21, "under_x" = 37, "under_y" = 16, "stock_x" = 9, "stock_y" = 12)
+	starting_attachment_types = list(/obj/item/attachable/scope/unremovable/tl127,
+									/obj/item/attachable/stock/tl127stock)
+burst_amount = 0
```
Comment (psykzz) [inline code/modules/projectiles/guns/rifles.dm:852]:
```suggestion
	burst_amount = 0
```

File: code/modules/projectiles/guns/rifles.dm [hunk 8]
```diff
@@ -815,3 +815,78 @@
...
+						/obj/item/attachable/suppressor,
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/bayonetknife,
+						/obj/item/attachable/compensator)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY|GUN_CAN_POINTBLANK
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO)
+	attachable_offset = list("muzzle_x" = 41, "muzzle_y" = 19,"rail_x" = 8, "rail_y" = 21, "under_x" = 37, "under_y" = 16, "stock_x" = 9, "stock_y" = 12)
+	starting_attachment_types = list(/obj/item/attachable/scope/unremovable/tl127,
+									/obj/item/attachable/stock/tl127stock)
```
Comment (psykzz) [inline code/modules/projectiles/guns/rifles.dm:851]: Lists should try to follow the following format, it helps speed up adding or removing instances and keeps git blames clean.

```suggestion
	starting_attachment_types = list(
		/obj/item/attachable/scope/unremovable/tl127,
		/obj/item/attachable/stock/tl127stock,
	)
```

File: code/modules/projectiles/gun_attachables.dm [hunk 9]
```diff
@@ -650,6 +650,13 @@ Defined in conflicts.dm of the #defines folder.
 /obj/item/attachable/scope/unremovable
 	flags_attach_features = ATTACH_ACTIVATION
 
+/obj/item/attachable/scope/unremovable/tl127
+	name = "T-45 rail scope"
+	aim_speed_mod = 0
+	wield_delay_mod = 0
+	attach_icon = "none"
+	desc = "A rail mounted zoom sight scope specialized for the T-127 sniper rifle . Allows zoom by activating the attachment. Use F12 if your HUD doesn't come back."
```
Comment (psykzz) [inline code/modules/projectiles/gun_attachables.dm:658]:
```suggestion
	desc = "A rail mounted zoom sight scope specialized for the T-127 sniper rifle. Allows zoom by activating the attachment. Use F12 if your HUD doesn't come back."
```

File: code/modules/projectiles/magazines/rifles.dm [hunk 10]
```diff
@@ -282,3 +282,14 @@
 	default_ammo = /datum/ammo/energy/plasma
 	max_rounds = 20
 	gun_type = /obj/item/weapon/gun/rifle/sectoid_rifle
+
+//-------------------------------------------------------
+//Marine magazine sniper, or the TL-127.
+/obj/item/ammo_magazine/rifle/chamberedrifle
+	name = "TL-127 bolt action rifle magazine"
+	desc = "A box magazine filled with 8.6x70mm rifle rounds for the TL-127."
+	caliber = "8.6×70mm"
```
Comment (psykzz) [inline code/modules/projectiles/magazines/rifles.dm:291]:
```suggestion
	caliber = "8.6x70mm"
```
is `×` used everywhere?

**Comment by Hughgent** [comment]
This should not have an 'unremoveable stock'

the sprite sheet it is a part of is wide enough to fit the entire weapon.

Please do the following.
1. Delete the stock 'attachment' from attachables as well as the weapon.
In theory, should someone care to make it, a rubber buttpad could be an appropriate stock attachment for a weapon like this. (or any stocked weapon)
2. Move the gun64 sprite to the right until the stock can fit.
3. Move the stock attachment sprite onto the weapon itself
4. if it is 'too far to the right' in game, add `pixel_x = -X` to the weapon object to shift it back left.

**Comment by Pariah919** [comment]
@Hughgent Its so it doesn't clog up the entire hand. Its fine as it is.

**Comment by Hughgent** [comment]
@Pariah919 incorrect. 

If it's 'covering up the hand' as you say, it should have hand cutouts for the on mob sprite.

@Kuro-ai

**Comment by Kuro-ai** [comment]
He means the obj sprite not mob

**Comment by Hughgent** [comment]
My requested change is not possible due to how the inventory displays items. When an item enters your on screen hand slot, the image displayed is always at pixel_x = 0 and pixel_y = 0.

File: code/modules/projectiles/guns/rifles.dm [hunk 11]
```diff
@@ -815,3 +815,80 @@
...
+	name = "\improper TL-127 bolt action rifle"
+	desc = "The TL-127 is the standard issue bolt action rifle used by the TGMC. Known for its long range accuracy and use by marksmen despite its age and lack of IFF. It has an irremoveable scope. Uses 8.6×70mm box magazines."
+	icon = 'icons/Marine/gun64.dmi'
+	icon_state = "tl127"
+	item_state = "tl127"
+	fire_sound = 'sound/weapons/guns/fire/tl127.ogg'
+	dry_fire_sound = 'sound/weapons/guns/fire/sniper_empty.ogg'
+	unload_sound = 'sound/weapons/guns/interact/m41a_unload.ogg'
+	reload_sound = 'sound/weapons/guns/interact/m41a_reload.ogg'
+	caliber = "8.6×70mm" //codex
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/rifles.dm:833]: Caliber is displayed weirdly. http://prntscr.com/tga8cs

File: code/modules/projectiles/guns/rifles.dm [hunk 12]
```diff
@@ -815,3 +815,80 @@
...
+	max_shells = 10 //codex
+	current_mag = /obj/item/ammo_magazine/rifle/chamberedrifle
+	type_of_casings = "cartridge"
+	attachable_allowed = list(
+						/obj/item/attachable/bayonetknife,
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/rifles.dm:838]:
```suggestion
```
Listed twice.

File: code/modules/projectiles/guns/rifles.dm [hunk 13]
```diff
@@ -815,3 +815,80 @@
...
+		return
+	if(COOLDOWN_CHECK(src, COOLDOWN_RACK_BOLT))
+		return
+	return rack_bolt(user)
+
+/obj/item/weapon/gun/rifle/chambered/proc/rack_bolt(mob/user)
+	to_chat(user, "<span class='notice'>You cycle the bolt of the [src], loading in a new round!</span>")
+	COOLDOWN_START(src, COOLDOWN_RACK_BOLT, rack_delay)
+	racked_bolt = TRUE
+	playsound(loc, rack_sound, 25, 1, 4)
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/rifles.dm:890]: Rack sound is not playing for some reason.

File: code/modules/projectiles/guns/rifles.dm [hunk 14]
```diff
@@ -815,3 +815,80 @@
...
+	. = ..()
+	if(!.)
+		return
+	if(!racked_bolt)
+		to_chat(user, "<span class='notice'>[src] does not have a round chambered!</span>")
+		return FALSE
+
+/obj/item/weapon/gun/rifle/chambered/unique_action(mob/user)
+	if(racked_bolt)
+		to_chat(user, "<span class='notice'>[src] already has a round chambered!</span>")
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/rifles.dm:880]: There are also no sounds when the round is already chambered.
```suggestion
		to_chat(user, "<span class='warning'>[src] already has a round chambered!</span>")
```

File: code/modules/projectiles/guns/rifles.dm [hunk 15]
```diff
@@ -815,3 +815,80 @@
...
+	var/rack_sound = "tl-127_bolt.ogg"
+	var/racked_bolt = TRUE
+	var/cooldown_time
+
+/obj/item/weapon/gun/rifle/chambered/able_to_fire(mob/user)
+	. = ..()
+	if(!.)
+		return
+	if(!racked_bolt)
+		to_chat(user, "<span class='notice'>[src] does not have a round chambered!</span>")
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/rifles.dm:875]:
```suggestion
		to_chat(user, "<span class='warning'>[src] does not have a round chambered!</span>")
```

File: code/modules/projectiles/guns/rifles.dm (hunk 11 above)
Comment (psykzz) [reply code/modules/projectiles/guns/rifles.dm:833]: `×` is a unicode character causing the issue, we should be replacing all of them with the normal x.

**Comment by Hathkar** [comment]
Did some testing, and it seems a bit overtuned (in my opinion).
Rate of fire is quite high, and you can slap a barrel charger on it with no loss in accuracy.
7.5 sunder also seems quite high considering the damage.

**Pull request merged**